src/hotspot/share/runtime/thread.cpp

```diff
@@ -3900,3 +3900,41 @@ void JavaThread::verify_cross_modify_fence_failure(JavaThread *thread) {
    report_vm_error(__FILE__, __LINE__, "Cross modify fence failure", "%p", thread);
 }
 #endif
+
+// Starts the target JavaThread as a daemon of the given priority, and
+// bound to the given java.lang.Thread instance.
+// The Threads_lock is held for the duration.
+void JavaThread::start_internal_daemon(JavaThread* current, JavaThread* target,
+                                       Handle thread_oop, ThreadPriority prio) {
+  MutexLocker mu(current, Threads_lock);
+
```

Comment:
Could you add an assert that target->os_thread() != NULL here?  Just to make sure everything checked.

Reply:
Done